runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -338,22 +341,29 @@ public void onVarbitChanged(VarbitChanged varbitChanged)
 		{
 			clientThread.invokeLater(this::updateTask);
 		}
-		else if (varbitId == Varbits.SLAYER_POINTS)
+		else if (varbitId == Varbits.SLAYER_POINTS
+			|| varbitId == Varbits.SLAYER_TASK_STREAK
+			|| varbitId == Varbits.SLAYER_WILDERNESS_TASK_STREAK
+			|| varbitId == Varbits.SLAYER_MASTER)
 		{
-			setProfileConfig(SlayerConfig.POINTS_KEY, varbitChanged.getValue());
-
-			// points is on a tooltip on the counter, so requires a rebuild if it changes
-			if (counter != null)
+			int varbitValue = varbitChanged.getValue();
+			switch (varbitId)
 			{
-				removeCounter();
-				addCounter();
+				case Varbits.SLAYER_POINTS:
+					setProfileConfig(SlayerConfig.POINTS_KEY, varbitValue);
+					break;
+				case Varbits.SLAYER_TASK_STREAK:
+					setProfileConfig(SlayerConfig.STREAK_KEY, varbitValue);
+					break;
+				case Varbits.SLAYER_WILDERNESS_TASK_STREAK:
+					setProfileConfig(SlayerConfig.WILDERNESS_STREAK_KEY, varbitValue);
+					break;
+				case Varbits.SLAYER_MASTER:
+					setProfileConfig(SlayerConfig.SLAYER_MASTER_KEY, varbitValue);
+					break;
```

Comment:
If this var value persists after a task is assigned, why do we save it to config?

Reply:
That's a good question. At the time of writing this PR, I very likely looked at the other varbits in the plugin (points, streak), which are also saved to the config, and assumed there was a good reason they were saved to the config. At least the points are also available after a task is completed as far as I can tell (tested on QSR), just like the slayer master.

If you'd like me to change this, and potentially do the same for points and/or streak, please let me know.

I'm not 100% sure if the streak is also always persistent. Currently on QSR it seems to be that way, but ``client.getVarbitValue(5671)`` shows that my wilderness task streak (non-QSR) is 0 while I'm on a regular task, which should be incorrect since I've done a bit of wilderness slayer (see screenshot of the original post) without ever turael skipping.

Reply:
I would assume it's a relic from before they sent all of the slayer information as varbits? It looks to be the case https://github.com/runelite/runelite/commit/1f6ae893d732cf3a022fe6d9662cd2a42ff2b217

Reply:
That seems to be the case indeed. Based on the brief discussion on Discord, I've removed saving the slayer master varb. During the next beta I might be able to test if the rest of the varbs and varps are persistent or not, potentially for a separate PR if this one would have already been merged (assuming we're even interested in removing them from the config).